cmd/thanos/receive.go

```diff
@@ -268,7 +268,19 @@ func runReceive(
 			return errors.Wrap(err, "parse limit configuration")
 		}
 	}
-	limiter, err := receive.NewLimiter(conf.writeLimitsConfig, reg, receiveMode, log.With(logger, "component", "receive-limiter"), conf.limitsConfigReloadTimer)
+	if conf.maxPendingGrpcWriteRequests > 0 {
+		level.Info(logger).Log("msg", "set max pending gRPC write request in limiter", "max_pending_requests", conf.maxPendingGrpcWriteRequests)
+	}
+	limiter, err := receive.NewLimiterWithOptions(
+		conf.writeLimitsConfig,
```

Comment:
consider reuse the writeLimitsConfig?

Reply:
I agree that it'd be ideal to have this config field in `writeLimitsConfig, but DB pods don't load `writeLimitsConfig` at all. Pantheon-writer pods load that config. I'll have to keep it this way.
The interface is not changed. `receive.NewLimiter()` stays the same. I added another function `receive.NewLimiterWithOptions()`.

Reply:
got it, that's fair, do you wanna add some unit tests for limiter to test the load shedding behavior?